### ISSUE TITLE
[BUG] Autocomplete renders loading component on clear

### ISSUE THREAD

**Issue by pcvonz** [magento-research/pwa-studio#958]
**Describe the bug**
When clearing the searchbar using the `x` button, the loading animation appears.

**To Reproduce**
1. Enter in a search query (greater than 3 characters)
2. Clear the query

**Expected behavior**
To immediately close the autocomplete component

**Screenshots**

![peek 2019-02-25 14-31](https://user-images.githubusercontent.com/6378569/53373388-1ec20a00-390a-11e9-8f4b-942ad7daec51.gif)

**Please complete the following device information:**
 - Device: PC
 - OS: Ubuntu 16.04
 - Browser Chromium
 - Version: 71
 - Magento Version: 2.3
 - NPM version `npm -v`: 6.4.1
 - Node Version `node -v`: 10.15.0

**Please let us know what packages this bug is in regards to:**
- [x] `venia-concept`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`

**Additional context**

This line in `autocomplete.js` is part of the issue:

https://github.com/magento-research/pwa-studio/blob/cf29cd684cc9740abfd8d0940a30a80f7c73b942/packages/venia-concept/src/components/SearchBar/autocomplete.js#L69

1. Clearing the query causes autocomplete to render again. Since `queryVariables.inputText` is debounced it searches for the query that was just cleared.
2. `queryVariables.inputText` is updated when the `debounceTime` expires causing another rerender which hides the `autocomplete` component.

**Possible solutions**

Checking against the actual `searchQuery` length. 
```js
        if (!autocompleteVisible || queryVariables.inputText.length < 3 || searchQuery.length === 0)
           null
```

**Comment by DmitryShmaliuk** [comment]
Please, assign to me.

**Comment by tjwiebell** [comment]
Fixed by #961
